osh/hub/osh_xmlrpc/scan.py

```diff
@@ -29,6 +40,59 @@
+    worker.channels.add(models.Channel.objects.get(name="default"))
+    worker.save()
+    print(worker.name)
+    print(worker.worker_key)
+    return worker
+
+def start_worker(worker):
+    # Generate /etc/osh/worker.conf through jinja template
+    worker_conf = render_to_string("worker.conf.j2", {"OSH_HUB_URL":settings.OSH_HUB_URL, "OSH_WORKER_KEY": worker.worker_key})
+    print(worker_conf)
+    # Create a temporary file and save the template
+    temporary_file = tempfile.NamedTemporaryFile()
+    temporary_file.write(bytes(worker_conf.strip(), 'utf-8'))
+    # Is this safe to assume this would always work?
+    temporary_file.flush()
+    # and transfer it to new worker.
+    # Copy worker.conf to centos user's home directory
+    # This requires the hub to have ssh access to the new worker
+    logger.info("scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} {} centos@{}:worker.conf".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, temporary_file.name, worker.name))
+    scp_exit_status = os.system("scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} {} centos@{}:worker.conf".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, temporary_file.name, worker.name))
+    if scp_exit_status != 0:
+        logger.error("Failed to copy worker.conf to new worker {}".format(worker.name))
+    # Delete the worker configs from hub
+    temporary_file.close()
+    logger.info("ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} centos@{} 'sudo mv worker.conf /etc/osh/worker.conf && sudo chown root:root /etc/osh/worker.conf'".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, worker.name))
+    # /etc/osh/hub/id_rsa.worker
+    move_worker_conf_status = os.system("ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} centos@{} 'sudo mv worker.conf /etc/osh/worker.conf && sudo chown root:root /etc/osh/worker.conf'".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, worker.name))
+    if move_worker_conf_status != 0:
+        logger.error("Failed to move worker.conf to new worker {}".format(worker.name))
+    # Start the worker after this is finished
+    logger.info("ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} centos@{} 'sudo systemctl restart osh-worker'".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, worker.name))
+    start_worker_status = os.system("ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {} centos@{} 'sudo systemctl restart osh-worker'".format(settings.RESALLOC_WORKER_SSH_PRIVATE_KEY, worker.name))
+    if start_worker_status != 0:
+        logger.error("Failed to start worker worker at {}".format(worker.name))
+
+
+def build_through_resalloc():
+    conn = ResallocConnection(settings.RESALLOC_SERVER)
+    ticket = conn.newTicket(settings.RESALLOC_WORKER_TAGS)
```

Comment:
@kdudka Where should this ticket id be stored? Shall we create a new field in `Task` table to have a reference to resalloc ticket id?

Reply:
Unless you are implementing a new feature of kobo, the information will need to stored on the OSH level.  Adding new fields (not used by kobo itself) to the `Task` table defined by kobo is difficult.  See https://github.com/release-engineering/kobo/pull/211#issuecomment-1690921525

Reply:
Sadly, modifying the `Task` model is not possible without breaking projects dependent on `kobo` with do not use or properly apply migrations.

Reply:
Would it be ok if I store it in the [comment](https://github.com/release-engineering/kobo/blob/8a704d50a8f56de0abfa63f3e383b5df129398fd/kobo/hub/models.py#L561) field of `Task` model? It's a hack, but that makes it easy to do a one to one mapping between task id and [worker events](https://github.com/openscanhub/openscanhub/blob/824beeba205847a66681f95503d7f457f77689de/osh/hub/osh_xmlrpc/worker.py#L39) without introducing a new field or table.

Reply:
Not really.  `comment` is there for users of OpenScanHub.  If they specify a comment with `osh-cli`, they expect to find the exact string in the field later on.